### PR TITLE
DrivAerML: T_max=36 + lr=4.8e-4 MSE-only (LR+schedule compound — HIGHEST PRIORITY)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis — HIGHEST PRIORITY EXPERIMENT

**Compound the two most important DM findings:** T_max=36 schedule (val=3.622%) and lr=4.8e-4 (full-eval TEST=4.117%, best in programme).

Your previous experiment (#3308) showed lr=4.8e-4 at T_max=30 gave val=3.657% (close to 3.622%) and full-eval TEST=4.117% (best ever, beating 4.218% by -2.4%). The T_max=36 schedule independently improved from 3.833% to 3.622%. These two improvements are likely orthogonal — lr changes the optimization step size, T_max changes the cycling frequency.

If they compound: val could drop below 3.6% AND test below 4.1%. This would be a major breakthrough.

**IMPORTANT:** Do NOT use `--ema-start-step`. The T_max=36 baseline (#3298) didn't use it, and we need a clean comparison.

## Current Baseline (DrivAerML)

- **val:** 3.622% at epoch 874 (T_max=36 MSE-only, lr=5e-4, chrome PR #3298)
- **full-eval TEST:** 4.218% (canute) / 4.117% (your lr=4.8e-4 at T_max=30, PR #3308)
- **SOTA target:** 3.71% AB-UPT (already beaten on val)

## Instructions to Student

```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 4.8e-4 \
  --cosine-t-max 36 \
  --grad-clip 0.5 \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --ema-decay 0.9995 \
  --ema-mode fixed \
  --no-compile-model \
  --save-checkpoint \
  --wandb_group dm-tmax36-lr48e4
```

After training, run full-eval on the best checkpoint:
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --load-checkpoint outputs/icml2026/drivaerml_senpai_transolver_best.pt \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --enable-fourier --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-eval-batches 0 --batch-size 1 --ema-decay 0.9995 --ema-mode fixed \
  --no-compile-model \
  --wandb_group dm-tmax36-lr48e4 --wandb_name "gojo/dm-tmax36-lr48e4-fulleval"
```

**What to report:**
1. Best `val/rel_l2_mean` and epoch
2. **Full-eval TEST** (critical — paper-facing number)
3. W&B run IDs for both training and full-eval
4. Stability through 600-min budget

**Target to beat:** val < 3.622% AND test < 4.117%

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT gojo:

## Results

### Training Run
| Metric | Baseline | This Run | Delta |
|--------|----------|----------|-------|
| Val (200-batch, primary) | 3.622% (ep874) | **3.521%** (ep850) | **-0.101%** ✅ |
| Full-eval TEST | 4.117% | **4.119%** | +0.002% (tied) |

- **Best epoch**: 850 (of 865 total, 600-min timeout)
- **Val broke below baseline at epoch 633** (3.615%) and continued descending steadily
- **Val at 3.5% range by epoch 820+**, with best 3.521% at ep850
- **Full-eval test**: 4.119% — essentially tied with baseline 4.117% (0.002% difference)

### W&B Runs
- Training: `5x2to2p8` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/5x2to2p8))
- Full-eval: `4z3ya8t9` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/4z3ya8t9))

### Training Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --optimizer adamw --lr 4.8e-4 --cosine-t-max 36 \
  --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 \
  --model-heads 8 --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --agent gojo --wandb-group dm-tmax36-lr48e4 --wandb-name "gojo/dm-tmax36-lr48e4"
```

### Full-eval Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml \
  --load-checkpoint outputs/icml2026/drivaerml_senpai_transolver_best.pt \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --enable-fourier --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-eval-batches 0 --batch-size 1 --ema-decay 0.9995 --ema-mode fixed \
  --no-compile-model \
  --agent gojo --wandb-group dm-tmax36-lr48e4 --wandb-name "gojo/dm-tmax36-lr48e4-fulleval"
```

### Peak Memory
~1.9 GB VRAM (single GPU)

### What Happened
The compound hypothesis (T_max=36 + lr=4.8e-4) **strongly beat the val baseline** with a 0.101% improvement (3.521% vs 3.622%). The model showed consistent, steady improvement throughout training, with the val metric descending from ~4.5% to 3.52% over 865 epochs. New bests were regularly set at cosine cycle troughs and mid-cycles, indicating the cosine annealing schedule was effectively exploring the loss landscape.

However, the **full-eval test metric (4.119%) essentially tied the baseline (4.117%)** rather than improving. The strong val improvement without corresponding test improvement suggests:
1. The 200-batch val subset may not fully represent the test distribution
2. The model may be finding configurations that perform well on the val subset but don't generalize proportionally to the held-out test set
3. The two improvements (T_max and LR) may not be fully orthogonal for generalization — T_max=36 helps val convergence while lr=4.8e-4 was tuned for test via a different baseline (T_max=30)

The val result is a clear new champion for the 200-batch eval metric, but the test result is flat relative to the baseline.

### Suggested Follow-ups
1. **Run with more epochs** — the model was still improving at timeout. With 874+ epochs (matching baseline), it might push test lower too
2. **Try T_max=36 with the exact baseline lr=5e-4** — the val baseline champion used lr=5e-4 with T_max=36. The lr=4.8e-4 was optimal for T_max=30. Maybe lr=5e-4 is better for T_max=36
3. **Investigate val/test divergence** — the large gap between val improvement and test flatness deserves analysis. Consider whether max_eval_batches=200 is a representative sample
4. **Try T_max=36 + lr=4.8e-4 with weight decay tuning** — the compound may benefit from adjusted regularization

**Comment by tcapelle** [comment]
ADVISOR: Merging as new DM val champion. Val=3.521% at ep850 beats 3.622% baseline by 0.101pp — the largest single val improvement we've seen on DrivAerML. Full-eval TEST=4.119%, essentially tied with the 4.117% reference.

**Important finding:** The full-eval val (4.456%) is dramatically different from the 200-batch val (3.521%), confirming ~0.9pp eval-subset bias. This matches the pattern seen with canute's T_max=38 run. All future paper-facing comparisons must use full-eval.

**What this means:** lr=4.8e-4 + T_max=36 MSE-only is the clear DM val champion. The compound of these two hyperparameters (each individually the best in their category) produces the best val result we've seen. The fact that test doesn't improve proportionally suggests the 200-batch eval subset and the test set probe different aspects of model quality.

Great work gojo. This run establishes the new DM baseline.
